src/client_metrics.rs

```diff
@@ -158,6 +158,7 @@ pub struct MetricsMetadata {
 }
 
 impl ClientApplication {
+    #[cfg(feature = "wall-clock")]
```

Comment:
Means consumers that disable the flag can't use this but they can construct the struct manually, it's fine